packages/block-editor/src/components/block-list/block-wrapper.js

```diff
@@ -107,6 +107,17 @@ const BlockComponent = forwardRef(
 				return;
 			}
 
+			// Don't move focus if it's outside the editor content. In the
+			// future, the check for this could look different, e.g. focus is
+			// outside the BlockList component, or outside the iframe.
+			if (
+				! wrapper.current
+					.closest( '[role="region"]' )
+					.contains( document.activeElement )
+			) {
+				return;
+			}
```

Comment:
This function is supposed to trigger only when a block gets selected IIRC, which means I'm not sure we need this check. Why is this function being called in that particular case, isn't the block already selected?

Reply:
Because the block is remounting :( Ideally, this shouldn't run if the block is remounting and the props didn't change. I'm not sure how to make that work. Do you have any ideas?

Reply:
Why the block is remounting?

Reply:
The block shouldn't be remounting when a color changes IMO

Reply:
That's what I think as well: https://github.com/WordPress/gutenberg/issues/21163#issuecomment-604416979

@jorgefilipecosta

Reply:
I'm not familiar enough with the useColors hook to figure out why and how it's remounting children.

Reply:
I see there's some rendering of cloned elements going on... 🤷‍♀️

Reply:
Looks like we can't do this particular fix anyway, it's breaking a lot of stuff. We need to find a way to not remount component nested in the color components.

Reply:
Another reason why I really do not like this way of magically updating the className and style props. ☹️